### PR TITLE
Add fillstyle option to linespecs

### PR DESCRIPTION
**Pull request by HorstBaerbel**
I hope I correctly put this into the linespecs category. Also removed some unused members from plotspecs. These are in linespecs. Probably a cp error...

### CODE REVIEW

File: sciplot/specs/plotspecs.hpp [hunk 1]
```diff
@@ -59,21 +59,6 @@ class plotspecs : public linespecs<plotspecs>
 
     /// The style of the plot (lines, points, linespoints) as a gnuplot formatted string (e.g., "with linespoints").
     std::string m_with;
-
```
Comment (allanleal) [inline sciplot/specs/plotspecs.hpp:62]: Hi @HorstBaerbel - just before I merge your PR, let me understand the following. With these data members now deleted, how are those plot properties represented (line type, line width, etc.)?

Comment (HorstBaerbel) [reply sciplot/specs/plotspecs.hpp:62]: As I understood it the values are output in [linespecs::repr()](https://github.com/sciplot/sciplot/blob/38fbb9e86364328ef531251cf95c947a524f9729/sciplot/specs/linespecs.hpp#L86) and all the members in plotspecs were unused. See[ plotspecs::repr()](https://github.com/sciplot/sciplot/blob/38fbb9e86364328ef531251cf95c947a524f9729/sciplot/specs/plotspecs.hpp#L84). All the private members I deleted are not output there. I suspect it was a copy-paste error. I might make sense to add a clang-tidy pass to your dev-env or the CI here to catch this...

Comment (HorstBaerbel) [reply sciplot/specs/plotspecs.hpp:62]: Also tests are lacking... ;)

**Comment by HorstBaerbel** [comment]
Added boxwidth command and change fillstyle enum class to lowercase as this seems to be the sciplot convention (I'm more of a camelCaseGuy though).

File: sciplot/specs/plotspecs.hpp (hunk 1 above)
Comment (allanleal) [reply sciplot/specs/plotspecs.hpp:62]: I'm trying to get a colleague to help with the devops.

**Comment by allanleal** [comment]
The `lowercase` (instead of `camelCase`) is to be closer to Gnuplot naming of commands. The core part of the lib can be more flexible with respect to this.

File: sciplot/specs/plotspecs.hpp (hunk 1 above)
Comment (HorstBaerbel) [reply sciplot/specs/plotspecs.hpp:62]: See my [GitHub workflows](https://github.com/HorstBaerbel/res2h/tree/master/.github/workflows) and actions (e.g. [this](https://github.com/HorstBaerbel/action-clang-tidy)) that use cmake + clang-tidy + clang-format for every commit / pull request. I use VS Code and its clang-format and clang-tidy plugins to check + format code on my local machine before commits.

**Comment by allanleal** [comment]
They look neat. We could adapt it to use here as well.

**Pull request merged**